apps/frontend/src/components/ui/ApplicationForm.tsx

```diff
@@ -0,0 +1,254 @@
+    }),
+    contact_email: z.string().email(),
+    is_sch_resident: z.boolean().optional(),
+    room_number: z
+      .union([
+        z.literal(0 && NaN),
+        z
+          .number()
+          .int()
+          .gte(201, { message: 'Ilyen szoba nem létezik' })
+          .lte(1816, { message: 'Ilyen szoba nem létezik' }),
+      ])
+      .optional()
+      .nullable()
+      .refine(
+        (data) => {
+          if (!data) return true;
+          const lastTwoDigits = data! % 100;
+          return lastTwoDigits >= 1 && lastTwoDigits <= 16;
+        },
+        { message: 'Cseles, de ilyen szoba nem létezik' }
+      ),
+    terms: z.boolean().refine((data) => data, { message: 'A szabályzatot el kell fogadnod' }),
+  })
+  .refine(
+    (data) => {
+      if (data.is_sch_resident) {
+        return data.room_number !== 0 && data.room_number !== undefined;
+      }
+      return true;
+    },
+    {
+      path: ['room_number'],
+      message: 'A szoba szám megadása kötelező, ha kolis vagy.',
+    }
+  );
+export default function ApplicationForm() {
+  const { toast } = useToast();
+  const router = useRouter();
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      nickname: 'Bujdi Bohoc',
+      contact_email: 'email@gmail.com',
+      is_sch_resident: false,
+      room_number: 0,
+      terms: false,
+    },
+  });
+
+  async function onSubmit(values: z.infer<typeof formSchema>) {
+    /*    toast({
+      title: 'a POST request has been sent:',
+      description: (
+        <pre className='mt-2 w-[340px] rounded-md bg-slate-950 p-4'>
+          <code className='text-white'>{JSON.stringify(values, null, 2)}</code>
+        </pre>
+      ),
+    });*/
+    try {
+      const response = await fetch('/api/submit', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(values),
+      });
+      //await response.json();
+      if (response.ok) {
+        toast({
+          title: 'Sikeres jelentkezés!',
+          description: 'Köszönjük, hogy kitöltötted a jelentkezési lapot!',
+        });
+        router.push('/');
+      } else {
+        toast({
+          title: 'Hiba történt!',
+          description: 'Valami nem stimmel a szerverrel, próbáld újra később!',
+          variant: 'destructive',
+        });
+      }
+    } catch (error) {
+      console.error('There was an error!', error);
+    }
+  }
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className='space-y-4 pb-16 md:mx-64 max-md:mx-8'>
+        <Card className='m-8 my-4'>
+          <CardHeader>
+            <CardTitle>Személyes adatok</CardTitle>
+            <CardDescription>Ellenőrízd személyes adataid, szükség esetén módosíts rajtuk!</CardDescription>
+          </CardHeader>
+          <CardContent className='md:grid-cols-4 md:grid gap-4'>
```

Comment:
Szóval ha itt a grid elől kiveszed az md-t akkor mobilon is megmarad a gap. Javít a vibeon jelentősen